linetools/isgm/abssystem.py

```diff
@@ -0,0 +1,202 @@
+""" Class for  absorption systems
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+# Python 2 & 3 compatibility
+try:
+    basestring
+except NameError:
+    basestring = str
+
+import numpy as np
+import warnings
+from abc import ABCMeta
+
+from astropy import units as u
+from astropy.table import QTable
+from astropy import constants as const
+from astropy.coordinates import SkyCoord
+
+from linetools.isgm.abscomponent import AbsComponent
+
+class AbsSystem(object):
+    """
+    Class for an absorption line system
+
+    Parameters
+    ----------
+    abs_type : str
+      Type of absorption system, e.g. MgII, DLA, LLS
+    radec : tuple or coordinate
+        RA/Dec of the sightline or astropy.coordinate
+    zabs : float
+      Absorption redshift
+    vlim : Quantity array (2)
+      Velocity limits of the system
+    zem : float, optional
+      Emission redshift of background source
+    NHI :  float, optional
+      Log10 of the HI column density
+    sigNHI :  np.array(2), optional
+      Log10 error of the HI column density (-/+)
+    name : str, optional
+      Name for the system
+
+    Attributes
+    ----------
+        abs_type : str
+        coord : SkyCoord
+            RA/Dec of the sightline
+        zabs : float
+          Absorption redshift
+        zem : float
+          Emission redshift of background source
+        vlim : Quantity array (2)
+          Velocity limits of the system
+        NHI :  float
+          Log10 of the HI column density
+        sigNHI :  np.array(2)
+          Log10 error of the HI column density (-/+)
+        MH :  float
+          Metallicity (log10)
+        name : str
+            Name of the system
+    """
+
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    def from_components(cls, components):
+        """Instantiate from a list of AbsComponent objects
+
+        Parameters
+        ----------
+        components : list
+          List of AbsComponent objects
+        """
+        # Check
+        if not isinstance(components,list):
+            raise IOError('Need a list of AbsComponent objects')
+        if not all(isinstance(x,AbsComponent) for x in components):
+            raise IOError('List needs to contain AbsLine objects')
```

Comment:
'List needs to contain AbsLine objects' -> 'List needs to contain only AbsComponent objects'